### ISSUE TITLE
gimp.lua "Edit with GIMP" does not work in DT 2.7 with windows

### ISSUE THREAD

**Issue by Mark-64** [darktable-org/lua-scripts#187]
Describe the bug
If I use "edit with GIMP", GIMP is launched, but the image is not opened.
GIMP gives a file not found error in its console, for example
failed: Could not open 'C:\Users\marco\AppData\Local\Temp\DSC03137.tif' for reading: No such file or directory
In fact the export image is not created in C:\Users\marco\AppData\Local\Temp, but it is created in the current film roll.
I guess the export image file is created in the wrong folder, or GIMP is called with a wrong parameter string.

Platform (please complete the following information):

    Windows 10
    DT version 2.7.0+1442~gca5f223e5

**Comment by supertobi** [comment]
@Mark-64 does it work with dt 2.6.x for you?

**Comment by wpferguson** [comment]
I'll take a look at it.  I currently don't have 2.7 on windows, so I'll have to install it.

**Comment by wpferguson** [comment]
@Mark-64 I just tried it and it worked for me.  Could you open a command prompt and run the following command please

c:\"program files\darktable\bin\darktable" -d lua > log.txt

Try and open a file with GIMP.  After you're done, please close darktable, then attach the log.txt file to this a comment and I'll see what it thinks is going on.

A couple of other questions...

Did GIMP open with no image or not start at all?

Did you use the file selector to specify the location of the GIMP executable?

Thanks,

Bill

**Comment by Mark-64** [comment]
@wpferguson 
Here is the log file 

```
25,303786 LUA "C:\Program Files\GIMP-2.10.10\gimp-2.10.10.exe" "C:\Users\marco\AppData\Local\Temp\DSC02990.tif" 
25,304786 LUA opened file

C:\Users\marco>"C:\Program Files\GIMP-2.10.10\gimp-2.10.10.exe" "C:\Users\marco\AppData\Local\Temp\DSC02990.tif"  
Creating rungimp logfile C:\Users\marco\AppData\Local\Temp\runGIMP-2.10.10-2019-06-13-18-35-21.log
Thu Jun 13 18:35:21 2019

Checking for Gimp .. 
Gimp exists! Continuing ... 
The gimp string is "gimp.exe" "C:\Users\marco\AppData\Local\Temp\DSC02990.tif" --verbose
25,372999 LUA result from windows command was 0
25,391004 LUA moving C:\Users\marco\AppData\Local\Temp\DSC02990.tif to E:\Immagini\WIP\Israel-2019\Capture/DSC02990.tif
25,532035 LUA importing file
25,560042 LUA DSC02990.ARW is a member
25,561042 LUA DSC02990.tif is a member
25,561042 LUA Already in group
```
Answers to you questions:

1) Gimp does start, but it doesn't open image. Instead, I can see the following error in the gimp Console

`Opening 'C:\Users\marco\AppData\Local\Temp\DSC02990.tif' failed: Error opening file C:\Users\marco\AppData\Local\Temp\DSC02990.tif: No such file or directory`

2) Yes I do use the file selector, as I'm using the Partha build of Gimp 2.10.10, which has a non standard path

**Comment by wpferguson** [comment]
So maybe I need to get the partha build and try that.

**Comment by Mark-64** [comment]
I have tried in my other laptop PC, also with Windows 10, DT 2.7 and standard Gimp 2.10.10 and it does work.
Something related to the Partha build then ?
I will investigate

**Comment by wpferguson** [comment]
Loaded Partha's build of gimp-2.10.10 and get the same behavior that you noted.  Then I tried the various different executables with the same problem.  Then I tried
"\program files\GIMP-2.10.10\gimp\bin\gimp.exe" \users\bill\AppData\Local\Temp\_7D_1834.tif 

and got an error complaining that libgexiv2-2.dll is missing from my computer.  So, Partha's gimp will run when clicked on, because it starts in the correct directory with all of it's dependencies, but it wont run from the command line.  When darktable starts gimp, it's using the command line behind the scenes.  So it tries to start gimp, gimp crashes, and the exported file just gets imported. If Partha's build is statically linked, then it will work or if it knows where to find it's dependencies.  You might want to file a bug report/feature request so that his distribution of gimp will start from the command line.

I downloaded and install gimp-2.10.10 from gimp.org and it works fine.
